syntax = "proto3";

package books;

service BookService {
    rpc GetBookById(GetBookRequest) returns (BookResponse);
    rpc GetAllBooks(GetAllBooksRequest) returns (BookListResponse);
}

message GetBookRequest {
    int32 id = 1;
}

message GetAllBooksRequest {}

message BookResponse {
    int32 id = 1;
    string title = 2;
    string author = 3;
    string publication_date = 4;
}

message BookListResponse {
    repeated BookResponse books = 1;
}
